detection_rules/rule.py

```diff
@@ -369,12 +369,18 @@ def __init__(self,
                  name: definitions.RuleName,
                  is_elastic_rule: bool,
                  note: Optional[definitions.Markdown] = None,
+                 interval: Optional[definitions.Interval] = None,
```

Comment:
Why pass this down vs pushing to the `extras` like the `from` value?

Reply:
My understanding is that the pass down is the preferred method (how it was done with note validation). From is an exception because we cannot pass down `from` because it is reserved it python.

Reply:
pass down provides type hinting